app/models/imo_delivery_detail.rb

```diff
@@ -0,0 +1,32 @@
+class ImoDeliveryDetail < ApplicationRecord
+  has_one :communication, as: :detailable
+
+  enum result: {
+    error: "error",
+    no_imo_account: "no_imo_account",
+    not_subscribed: "not_subscribed",
+    read: "read",
+    sent: "sent"
+  }
+
+  after_create :update_authorization, if: proc {|detail| detail.result.in? ["not_subscribed", "no_imo_account"] }
```

Comment:
Move guard to method to more tightly couple the permitted values to the functionality.